api/storage/elasticsearch/elasticsearch.go

```diff
@@ -327,6 +336,63 @@ func (sm *StorageManager) GetResources(resourceType string, executionID string,
 	return resources, nil
 }
 
+// GetResourceTrends return resource data
+func (sm *StorageManager) GetResourceTrends(resourceType string, filters map[string]string) ([]storage.ExecutionCost, error) {
+	// Per resource trends, filters should take care of granularity (per resource: Data.ResourceID, Data.Region, Data.Metric -> Data.PricePerMonth)
+	var resources []storage.ExecutionCost
+	var mustNotQuery []elastic.Query
+
+	// Must
+	mustQuery := sm.getDynamicMatchQuery(filters, "and")
+	mustQuery = append(mustQuery, elastic.NewMatchQuery("ResourceName", resourceType).Operator("and"))
+
+	// Unsupported Types - MustNot
+	mustNotQuery = append(mustNotQuery, elastic.NewMatchQuery("EventType", "service_status").Operator("and"))
+	mustNotQuery = append(mustNotQuery, elastic.NewMatchQuery("ResourceName", "aws_iam_users").Operator("and"))
+	mustNotQuery = append(mustNotQuery, elastic.NewMatchQuery("ResourceName", "aws_elastic_ip").Operator("and"))
+	mustNotQuery = append(mustNotQuery, elastic.NewMatchQuery("ResourceName", "aws_lambda").Operator("and"))
+	mustNotQuery = append(mustNotQuery, elastic.NewMatchQuery("ResourceName", "aws_ec2_volume").Operator("and"))
+
+	queryBuilder := elastic.NewBoolQuery().MustNot(mustNotQuery...).Must(mustQuery...)
+	searchResult, err := sm.client.Search().
+		Query(queryBuilder).
+		Pretty(true).
+		Size(100).
+		SortBy(elastic.NewFieldSort("Timestamp").Desc()).
+		Aggregation("executions", elastic.NewTermsAggregation().Field("ExecutionID").OrderByKeyAsc(). // Aggregate by ExecutionID
+														SubAggregation("monthly-cost", elastic.NewSumAggregation().Field("Data.PricePerMonth"))). // Sub aggregate and sum by Data.PricePerMonth per bucket
+		Do(context.Background())
+
+	if err != nil {
+		log.WithError(err).Error("elasticsearch query error")
+		return resources, err
+	}
+
+	executions, found := searchResult.Aggregations.Terms("executions")
+	if found {
+		for _, ppm := range executions.Buckets {
+			executionId := ppm.Key.(string)
+			monthlyAgg, _ := ppm.Aggregations.Sum("monthly-cost")
+
+			// Extract the timestamp from the ExecutionID
+			executionText := strings.Split(executionId, "_")
```

Comment:
We have a same login in function `GetExecutions` what do you say to merge them into one function?

Reply:
nice catch, I'll take care of it.

Reply:
fixed.